### ISSUE TITLE
Autocomplete stops showing the List when clicking in the input field after adding a Mat-Chip

### ISSUE THREAD

**Issue by vicmarcal** [angular/components#17292]
#### Reproduction

Use StackBlitz to reproduce your issue: 
Official Demo shows this issue.
https://run.stackblitz.com/api/angular/v1?file=app%2Fchips-autocomplete-example.ts

This example starts showing a <mat-chip>Lemon in the autocomplete component.

Steps to reproduce:
1. Click in the Input of Autocomplete, next to the Lemon matchip
2. It properly shows the List associated
3. Click in an Option of the List, i.e "Apple". The mat-chip is added next to "Lemon".
4. Click in the Input of Autocomplete. Now it does not show the "List", as you would expect and as it did in step 2, which breaks the UIX.
5. Click outside the component (it loses the Focus)
6. Click again the Input field: it receives the Focus and the List is again shown.
 
#### Expected Behavior

What behavior were you expecting to see?
Users expect that clicking in the Input field would show them the List of options again, as it did in the very beginning without needing to use the "hackish" click out-and-in.

#### Actual Behavior

What behavior did you actually see?
If a Mat-chip is added, the user needs to focus out and then click to set the Focus in, in order to have the List shown.

#### Environment

- Angular: 8.1.0
- CDK/Material: Material 8.0.2
- Browser(s): Chrome, Safari, Firefox
- Operating System (e.g. Windows, macOS, Ubuntu): Windows, Ubuntu, MacOS

**Comment by vicmarcal** [comment]
As an ugly fix, you can disable and enable the Control at the end of the "selected" method:
`formControl.disable();
  formControl.enable();`
as you can see in this proof of concept: 
https://angular-cp6d1k-h8up2b.stackblitz.io
However I bet this will expose some counterbacks with advanced user navigation probably.
Hope it helps!

**Comment by crisbeto** [comment]
I'm guessing it's because the list only shows up on focus and focus technically hasn't left the input. There's a PR to make it also open on clicks https://github.com/angular/components/pull/16020.

**Comment by vicmarcal** [comment]
Yup. That's my guess too "(...) the user needs to focus out and then click to set the Focus in, in order to have the List shown."
So a Focus issue to me. I didn't find a setFocus method for FormControls, so I decided to use the disable/enable which seemed to unset focus.

**Comment by crisbeto** [comment]
It might be simpler to call `blur` on the input.

**Comment by francittadini** [comment]
> It might be simpler to call `blur` on the input.

This worked for me, good idea :)

**Comment by GitBrainsStudio** [comment]
> It might be simpler to call `blur` on the input.

how it make?

**Comment by cfgordo-OSS2D** [comment]
> It might be simpler to call `blur` on the input.

Problem with blur and with formControl.disable(); formControl.enable(); (you can try both in stackblitz), you lose focus on the key entry for the autocomplete, then the usage becomes clunky. If you are typing and then combine with mouse clicks to select (not a totally weird combo because it is quicker to type, shorten the list, then click with a mouse then to scroll) you lose focus on select and have the same problem.

**Comment by hiepxanh** [comment]
``` //.ts

const chipElement = document.querySelector('#chip-list');
chipElement.blur();
setTimeout(() => { chipElement.focus() }, 1);
```
help resolve the problem for me.

**Comment by almcaffee** [comment]
For anyone still needing this...

`  @ViewChild('inputName') inputName: ElementRef<HTMLInputElement>;

  selected(event: MatAutocompleteSelectedEvent): void {
    ....
    this.inputName.nativeElement.blur();
    ....
  }
`

Hope this helps tool a while to get it working smoothly today

**Comment by Koniouchine** [comment]
> For anyone still needing this...
> 
> ```
>   @ViewChild('inputName') inputName: ElementRef<HTMLInputElement>;
> 
>   selected(event: MatAutocompleteSelectedEvent): void {
>     ....
>     this.inputName.nativeElement.blur();
>     ....
>   }
> ```
> 
> Hope this helps took a while to get it working smoothly today
> 
> **Edit: Heres a snippet of the template...**
> 
> ```
> <mat-form-field>
>       <mat-label>{{ 'views.user.permissions.additional_data_center' | translate }}</mat-label>
>       <mat-chip-list #chipList aria-label="{{ 'views.user.permissions.additional_data_center' | translate }}">
>           <mat-chip *ngFor="let d of dataCenters | idFilter:'id':selectedDataCenters:5" 
>               [selectable]="true"
>               [removable]="true" 
>               (removed)="remove(d)">
>               {{d.name}}
>               <mat-icon matChipRemove>cancel</mat-icon>
>           </mat-chip>
>           <input #dcInput
>                   [matChipInputFor]="chipList"
>                   [matAutocomplete]="auto"
>                   [matChipInputSeparatorKeyCodes]="separatorKeysCodes"
>                   [matChipInputAddOnBlur]="true"
>                   (matChipInputTokenEnd)="add($event)">
>           <mat-autocomplete #auto="matAutocomplete" (optionSelected)="selected($event)">
>               <mat-option *ngFor="let dc of filterDataCenters" [value]="dc.id">
>               {{dc.name}}
>               </mat-option>
>           </mat-autocomplete>
>           <span matSuffix *ngIf="selectedDataCenters.length > 5">+{{selectedDataCenters.length - 5 }} </span>
>       </mat-chip-list>
>   </mat-form-field>
> ```

This was a clear solution to the chip list autocomplete not working properly after selection. Even the current documentation has this bug. 
Thanks for the help!

**Comment by almcaffee** [comment]
@Koniouchine .. glad I could help.

**Comment by 128keaton** [comment]
I made this little directive to "fix" this issue
Currently, it'll show after adding from the list or removing from the list

https://gist.github.com/128keaton/a67c3fe14d70dd3becd4628f56b18e34

**Comment by atilaoliveirajr** [comment]
> I made this little directive to "fix" this issue
> Currently, it'll show after adding from the list or removing from the list
> 
> https://gist.github.com/128keaton/a67c3fe14d70dd3becd4628f56b18e34
> 
> Just bind it to the `mat-form-field` containing everything
> ![Screen Shot 2021-04-14 at 4 24 44 PM](https://user-images.githubusercontent.com/1402918/114781262-11f90e00-9d3e-11eb-88d2-19c1304299ec.png)

Worked like a charm! Perfect solution. Thank you!
